### ISSUE TITLE
A Guided Raodmap for Learning Programming(And Other Things)

### ISSUE THREAD

**Issue by asif10388** [DevArifHossain/saas-idea-validator#2]
A platform where learners can learn programming by following interactive screen casts and take part in learning by doing unique projects and implementing creative solutions to problems faced in the real world both in startups and corporate companies. They will also learn to read other people's code to understand how different solutions work and how to work with a new code base. This ability is crucial in the job world as every new and entry level engineer is given an unfamiliar code base to work with when they first join. The platform will also provide one to one sessions and collaborative learning. It's going to be a platform that focuses on students being able to solve problems for themselves rather than follow tutorials and receive certificates to only move onto another course.

**Comment by DevArifHossain** [comment]
So users will learn programming by doing projects on screen casts??

**Comment by asif10388** [comment]
Users will learn the fundamentals of certain technologies (Python, React, Algorithms) via screen casts recorded by experts. Then based on what they learn from the screen casts, and their expertise level, they will be given a curated list of projects to work on, solve or collaborate on. They can customize the difficulty level of and the uniqueness of the projects. So no two tasks will be the same.

**Comment by DevArifHossain** [comment]
Nice...
So what is understood is, It's kind of like Codeacademy but the difference is the tutorials & projects will be different for different users?

**Comment by asif10388** [comment]
Yes. It will be geared towards ensuring every user gets a unique learning experience and when applying for jobs, they can stand out with their resumes.

**Comment by DevArifHossain** [comment]
Cool. So your target customer is people show wants to get into tech jobs? 
and how will the pricing work?

**Comment by asif10388** [comment]
For starters I am targeting people who wants to learn the fundamentals. Later I will add career paths and people can choose to either learn because of their passion, or to understand difficult university topics, or to follow a rigorous yet entertaining roadmap and get a job.

For the pricing, I haven't really decided on the pricing model yet. Hopefully I will have one ready when I'm done building. But the platform will have a free tier.
